Modules/Office365DSC/DSCResources/MSFT_TeamsMemberSettings/MSFT_TeamsMemberSettings.psm1

```diff
@@ -0,0 +1,230 @@
+        AllowCreateUpdateRemoveConnectors = $null
+        Ensure                            = "Absent"
+        GlobalAdminAccount                = $GlobalAdminAccount
+    }
+
+    Write-Verbose -Message "Getting Team member settings for $TeamName"
+
+    $team = Get-Team |  Where-Object {$_.DisplayName -eq $TeamName}
+    if ($null -eq $team)
+    {
+        throw "Team with Name $TeamName doesnt exist in tenant"
+    }
+
+    $teamMemberSettings = Get-TeamMemberSettings -GroupId $team.GroupId -ErrorAction SilentlyContinue
+    if ($null -eq $teamMemberSettings)
+    {
+        Write-Verbose "The specified Team doesn't exist."
+        return $nullReturn
+    }
+
+    Write-Verbose "Team member settings for AllowCreateUpdateChannels = $($teamMemberSettings.AllowCreateUpdateChannels)"
+    Write-Verbose "Team member settings for AllowDeleteChannels = $($teamMemberSettings.AllowDeleteChannels)"
+    Write-Verbose "Team member settings for AllowAddRemoveApps = $($teamMemberSettings.AllowAddRemoveApps)"
+    Write-Verbose "Team member settings for AllowCreateUpdateRemoveTabs = $($teamMemberSettings.AllowCreateUpdateRemoveTabs)"
+    Write-Verbose "Team member settings for AllowCreateUpdateRemoveConnectors = $($teamMemberSettings.AllowCreateUpdateRemoveConnectors)"
+
+
+    return @{
+        TeamName                          = $TeamName
+        AllowCreateUpdateChannels         = $teamMemberSettings.AllowCreateUpdateChannels
+        AllowDeleteChannels               = $teamMemberSettings.AllowDeleteChannels
+        AllowAddRemoveApps                = $teamMemberSettings.AllowAddRemoveApps
+        AllowCreateUpdateRemoveTabs       = $teamMemberSettings.AllowCreateUpdateRemoveTabs
+        AllowCreateUpdateRemoveConnectors = $teamMemberSettings.AllowCreateUpdateRemoveConnectors
+        Ensure                            = "Present"
+        GlobalAdminAccount                = $GlobalAdminAccount
+    }
+
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $TeamName,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowCreateUpdateChannels,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowDeleteChannels,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowAddRemoveApps,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowCreateUpdateRemoveTabs,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowCreateUpdateRemoveConnectors,
+
+        [Parameter()]
+        [ValidateSet("Present", "Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    if ('Absent' -eq $Ensure)
+    {
+        throw "This resource cannot delete Managed Properties. Please make sure you set its Ensure value to Present."
```

Comment:
Update description of error to be reflective of the resource

Reply:
done